include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1034,7 +1034,8 @@ namespace internal
       touch_count.reinit(transfer.partitioner_fine);
 
       for (const auto i : transfer.level_dof_indices_fine)
-        touch_count.local_element(i) += 1;
+        if (i != numbers::invalid_dof_index)
+          touch_count.local_element(i) += 1;
```

Comment:
I am surprised that we can end up with `numbers::invalid_unsigned_int` here. Is this documented somewhere?

Reply:
Why is the number `invalid_dof_index` here in the first place and not `invalid_unsigned_int`? It seems that `level_dof_indices_fine` is a vector of `unsigned int` because it reflects indices in MPI-local index space https://github.com/dealii/dealii/blob/dd14d069d2c8b694474d9fe1fdb2a2190e9d7e33/include/deal.II/multigrid/mg_transfer_global_coarsening.h#L426 . So in line 1036 it might help to spell out the right type.

@drwells to answer your question: We keep these numbers for all SIMD lanes. In case only some SIMD lanes are populated, this number indicates that no useful information is stored.